### ISSUE TITLE
Accessibility features are not correctly marked/missing in the select element 

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#1490]
### Provide a general summary of the issue here

The <sl-select> element has some erroneously marked accessibility features and is missing some. Currently the correct properties are marked between two elements (sl-select) and (sl-select-button).

- Label is connected to the wrong element
- Hint/error text is connected to the wrong element
- Required is in the wrong element
- Aria-placeholder is used in an element/role which does not support it  and aria-label in an element, which does not need it
- Aria-controls and aria-expanded="false" are missing

### 🤔 Expected Behavior?

When using a screen reader and focusing on the select element, the element should announce it's label, state and any other expected features to a screen reader. This means, that the element's label is read, any hints/error texts connected are read, the element is correctly identified and it's state is all interpreted by a screen reader at once. 

In this case, a screen reader should announce, that this is a combobox button, it's label, it's state (closed/opened), the selected element and if applicable, the required state and any connected hint or error texts. 

### 😯 Current Behavior

Currently some screen readers only announce, that the element is a combobox, it's currently selected element and that it opens a list.

(Some mobile screen readers do some fancier grouping, and can interpret these correctly)

### 💁 Possible Solution

**Reason for the bug:**
Currently the element’s properties are shared between the sl-select and sl-select-button element. In actuality, the screen reader focuses to the sl-select-button (due to the host delegating focus to children in shadowroot). Specifically, the “required” attribute and aria-describedby are present in the sl-select tag, which is not focused. 
<img width="502" alt="sl-select marked" src="https://github.com/user-attachments/assets/d70b6471-feda-4a1a-bf3e-5a1eca522b71">

Another element, that is pointing to the wrong element is the label of the element, which is in the sl-label tag.
<img width="502" alt="sl-select label" src="https://github.com/user-attachments/assets/59e3a490-492a-46d0-91ce-7f36e223190d">

**Solution:**
<img width="502" alt="sl-select changes" src="https://github.com/user-attachments/assets/03ba20d0-ddea-4eba-9b4b-ff71e50e2801">
- Move the required and aria-description from the sl-select to sl-select-button, and add the aria-expanded attribute (In green)
- Connect the label to the correct element using id (In red)
- Connect aria-controls to the correct element using “id”, required adding an id to the element with role=”listbox”. (In blue) 
- Remove aria-label from div with role=”listbox” and aria-placeholder from sl-select-button as they are redundant 

### 🖥️ Steps to Reproduce

1. Open a page with the select element (e.g. [Select required](Open https://storybook.sanomalearning.design/iframe.html?id=form-select--required&viewMode=story)
2. Start a screen reader (e.g. NVDA for Windows)
3. Focus the select element (for NVDA use tab)

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 10

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
```[tasklist]
### Tasks
- [ ] investigate the best solution
- [x] use aria-labelby for label element ?
- [x] form control needs to be linked to label
- [ ] solution within combo box can be reused in select
- [x] Test with NVDA and Talkback (screen reader for Android, ask Anna Karon to test)
```

**Comment by anqaka** [comment]
I reviewed the select component and tested it with VO:

1. Label should be attached to the select element (<sl-select-button role="combobox" ...>) with aria-labelledby  attribute with the label id value. Currently, the name of the combobox is empty
2. Add aria-controls attribute with listbox id value (aria-controls=”<listbox-id>”)
3. aria-expanded attribute is visible only when the value is "true"
4. Error message (<sl-error>) should be attached to the <sl-select-button role="combobox" ...>, with aria-describedby attribute. Currently is attached to the <sl-select>

My suggestions coincide with those in the comment above.
Tested components on storybook:
1. [select required](https://storybook.sanomalearning.design/?path=/story/form-select--required)
2. [Select basic](https://storybook.sanomalearning.design/?path=/story/form-select--basic)

**Comment by anna-lach** [comment]
Waiting for feedback from Anna.
